django/contrib/admindocs/utils.py

```diff
@@ -145,93 +146,53 @@ def default_reference_role(name, rawtext, text, lineno, inliner, options=None, c
-    for group_pattern, group_name in group_pattern_and_name:
-        pattern = pattern.replace(group_pattern, group_name)
-    return pattern
+# Match unescaped parenthesis in a regex, simplify groups and then clean.
+parenthesis_matcher = re.compile(r'\\.|([()])')
+groups_matcher = re.compile(r'^\((?:\?(?:([iLmsux:])|P(<\w+>)|([P#=!<(]))|(?#empty))(.*)\)$')
+cleanup_matcher = re.compile(
+    r'(\\0[0-9]{0,3})|\\([^1-9AbBdDsSwWZafnrt])|\\[1-9][0-9]?|\\[AbBdDsSwWZafnrt]|[.^$*+?|]|\[[^]]*\]|\{[^}]*\}')
 
 
-def replace_unnamed_groups(pattern):
+def simplify_regex(pattern):
     r"""
-    Find unnamed groups in `pattern` and replace them with '<var>'. E.g.,
-    1. ^(?P<a>\w+)/b/(\w+)$ ==> ^(?P<a>\w+)/b/<var>$
-    2. ^(?P<a>\w+)/b/((x|y)\w+)$ ==> ^(?P<a>\w+)/b/<var>$
+    Clean up urlpattern regexes into something more readable by humans. For
+    example, turn "^(?P<sport_slug>\w+)/athletes/(?P<athlete_slug>\w+)/$"
+    into "/<sport_slug>/athletes/<athlete_slug>/".
     """
-    unnamed_group_indices = [m.start(0) for m in unnamed_group_matcher.finditer(pattern)]
-    # Indices of the start of unnamed capture groups.
-    group_indices = []
-    # Loop over the start indices of the groups.
-    for start in unnamed_group_indices:
-        # Handle nested parentheses, e.g. '^b/((x|y)\w+)$'.
-        unmatched_open_brackets, prev_char = 1, None
-        for idx, val in enumerate(list(pattern[start + 1:])):
-            if unmatched_open_brackets == 0:
-                group_indices.append((start, start + 1 + idx))
-                break
-
-            # Check for unescaped `(` and `)`. They mark the start and end of
-            # a nested group.
-            if val == '(' and prev_char != '\\':
-                unmatched_open_brackets += 1
-            elif val == ')' and prev_char != '\\':
-                unmatched_open_brackets -= 1
-            prev_char = val
-
-    # Remove unnamed group matches inside other unnamed capture groups.
-    group_start_end_indices = []
-    prev_end = None
-    for start, end in group_indices:
-        if prev_end and start > prev_end or not prev_end:
-            group_start_end_indices.append((start, end))
-        prev_end = end
-
-    if group_start_end_indices:
-        # Replace unnamed groups with <var>. Handle the fact that replacing the
-        # string between indices will change string length and thus indices
-        # will point to the wrong substring if not corrected.
-        final_pattern, prev_end = [], None
-        for start, end in group_start_end_indices:
-            if prev_end:
-                final_pattern.append(pattern[prev_end:start])
-            final_pattern.append(pattern[:start] + '<var>')
-            prev_end = end
-        final_pattern.append(pattern[prev_end:])
-        return ''.join(final_pattern)
-    else:
-        return pattern
+    indices = [m.start(1) for m in parenthesis_matcher.finditer(pattern) if m.start(1) != -1]
```

Comment:
Could you add some comments to this function similar to how the old code is documented?